lib/datadog/tracing/sampling/span/rule_parser.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'datadog/tracing/sampling/span/ext'
+require 'datadog/tracing/sampling/span/matcher'
+require 'datadog/tracing/sampling/span/rule'
+
+module Datadog
+  module Tracing
+    module Sampling
+      module Span
+        # Converts user configuration into {Datadog::Tracing::Sampling::Span::Rule} objects,
+        # handling any parsing errors.
+        module RuleParser
+          class << self
+            # Parses the provided JSON string containing the Single Span
+            # Sampling configuration list.
+            # In case of parsing errors, `nil` is returned.
+            #
+            # @param rules [String] the JSON configuration rules to be parsed
+            # @return [Array<Datadog::Tracing::Sampling::Span::Rule>] a list of parsed rules
+            # @return [nil] if parsing failed
+            def parse_json(rules)
+              begin
+                list = JSON.parse(rules)
+              rescue JSON::ParserError => e
+                Datadog.logger.warn("Error parsing Span Sampling Rules \"#{e}\" for rules: #{rules}")
```

Comment:
I intentionally removed `Location: #{Array(e.backtrace).first}`: for JSON parsing the message (`e.to_s`) has a very clear error for all errors I tested. This message, alongside the original JSON string (`rules`) was always enough to have actionable information to fix the issue.

The one issue I found was if a non-String as passed to parse, and thus relaxed the `JSON::ParserError => e` to `=> e` in a future PR.

I'll backport that change to this PR.

Reply:
Perhaps worth leaving a quick comment there `# ParseError already includes a lot of info so I didn't see a need for e.class/e.backtrace` or something, otherwise someone may end up randomly adding it as part of a consistency "cleanup" later (and that someone may be me in a few months having forgotten this discussion :P).